lxd/storage/drivers/driver_btrfs_volumes.go

```diff
@@ -905,28 +910,38 @@ func (d *btrfs) RefreshVolume(vol VolumeCopy, srcVol VolumeCopy, refreshSnapshot
 		return err
 	}
 
+	revert.Add(func() {
+		_ = d.DeleteVolumeSnapshot(srcSnap, op)
+	})
+
 	// Transfer temporary snapshot to target; this creates a new snapshot for target.
 	parentSnap, err := srcVol.NewSnapshot(srcSnapshotsAll[len(srcSnapshotsAll)-1])
 	if err != nil {
 		return err
 	}
 
-	err = transfer(srcSnap, vol.Volume, parentSnap)
+	targetSnap, err := vol.NewSnapshot(snapUUID)
 	if err != nil {
 		return err
 	}
 
-	err = d.DeleteVolumeSnapshot(srcSnap, op)
+	err = transfer(srcSnap, vol.Volume, parentSnap)
 	if err != nil {
+		// Cleanup the potentially left over temporary snapshot transferred to target.
+		revertErr := d.setSubvolumeReadonlyProperty(targetSnap.MountPath(), false)
```

Comment:
Does the readonly property has to be set to `false` in order to allow its deletion? Or is it a check to see if the subvol exists on the target?

Reply:
I think it needs to be set to not be readonly to remove